Feature/ui revamp countries list

m1awicki opened the pull request:
Compact list of countries (iso-2 code + flag image) with information about daily increase in cases, deaths and recoveries.
Country item can be expanded to show bit more details and mark country as followed.
Followed countries are pinned to the top of the list and automatically expanded.

m1awicki committed:
Fix date issues

m1awicki committed:
Store URL of country flag

m1awicki committed:
Add nw countries list with collapsible details

m1awicki committed:
Mark string as not translantable

MDikkii commented on app/src/main/java/com/mobilecoronatracker/data/persistence/entity/Country.kt:
```diff
@@ -4,6 +4,7 @@ import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.Index
 import androidx.room.PrimaryKey
+import java.sql.Blob
```
Please remove unused import

m1awicki replied:
done

MDikkii commented on app/src/main/res/layout/fragment_countries_list.xml:
```diff
@@ -6,49 +6,46 @@
             android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:background="@drawable/rounded_rect_bg"
+            android:background="@color/colorPrimary"
```
I would vote for a different background. Maybe surface background (primaryDark?). And icon should be white IMO.

m1awicki replied:
done

MDikkii commented on app/src/main/res/layout/item_country_report.xml:
```diff
@@ -6,181 +6,282 @@
+                android:id="@+id/deaths_delta"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:gravity="center_vertical|right"
```
Please use start / end

m1awicki replied:
done

MDikkii commented on app/src/main/res/layout/item_country_report.xml:
```diff
@@ -6,181 +6,282 @@
+                android:id="@+id/recovered_delta"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginRight="20dp"
```
Please use start / end

m1awicki replied:
done

MDikkii commented on app/src/main/res/layout/item_country_report.xml:
```diff
@@ -6,181 +6,282 @@
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginRight="20dp"
+                android:gravity="center_vertical|right"
```
Please use start / end

m1awicki replied:
done

MDikkii commented on app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt:
```diff
@@ -0,0 +1,63 @@
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) =
+        holder.bind(deltas[position])
+
+    private fun setDetailsVisible(detailsContainer: ItemCountryReportBinding, visible: Boolean) {
```
`detailsContainer` might be hard to understand that it is binding

m1awicki replied:
Yep. Missed that one during refactoring.

MDikkii commented on app/src/main/java/com/mobilecoronatracker/ui/countrieslist/CountriesListAdapter.kt:
```diff
@@ -0,0 +1,63 @@
+            if (item.countryReport.flagPath.isNotEmpty()) {
+                binding.countryFlag.load(item.countryReport.flagPath)
+            }
+            binding.alwaysVisible.setOnClickListener {
```
Maybe whole layout should be clickable to make hiding easier?

m1awicki replied:
On the other hand it is easier to accidentally collapse it. Are you sure it is better approach to make it clickable as a whole?

MDikkii replied:
As agreed on the other channel, let's keep only 'always visible' part clickable.

MDikkii commented on app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt:
```diff
@@ -88,8 +66,15 @@ class FragmentCountriesReports : Fragment() {
+            adapter.deltas = it
             adapter.notifyDataSetChanged()
         })
+        viewModel.toggleCollapse.observe(viewLifecycleOwner, Observer {
```
Which item collapsible_container will be called here?

m1awicki replied:
That was leftover, missed during refactoring. Removed.

MDikkii commented on app/src/main/java/com/mobilecoronatracker/ui/countrieslist/FragmentCountriesList.kt:
```diff
@@ -88,8 +66,15 @@ class FragmentCountriesReports : Fragment() {
     private fun bindObservers() {
         viewModel.countryReports.observe(viewLifecycleOwner, Observer {
-            adapter.reports = it
+            adapter.deltas = it
```
I don't like `deltas` as a name for  a whole country reports :D

m1awicki replied:
Me either xD

m1awicki committed:
Improvements

Removed obsolete code and improved variables names.
Improved layout for countries list item.

m1awicki committed:
Fix missing underline in SearchView

m1awicki committed:
Bring back the menu
